git/repo/base.py

```diff
@@ -969,7 +970,18 @@ def _clone(cls, git, url, path, odb_default_type, progress, multi_options=None,
             handle_process_output(proc, None, progress.new_message_handler(), finalize_process, decode_streams=False)
         else:
             (stdout, stderr) = proc.communicate()
-            log.debug("Cmd(%s)'s unused stdout: %s", getattr(proc, 'args', ''), stdout)
+            cmdline = getattr(proc, 'args', '')
+            uri = cmdline[-2]
```

Comment:
I think there is at least two arguments here, so `cmdline[-2]` will never fail. However, it would be great to only run the url parsing code if `cmdline[1]` is actually `clone`.

Reply:
In a second step, one could consider other `git` commands as well or maybe even search arguments for URLs with password, allowing to fetches and pulls as well along with less common invocations.

Reply:
You're right this is a more general problem. I think I've move the code to utils and use it also in cmd.execute.

Reply:
Also I'm parsing any elem of the command line just to be sur